### PR TITLE
Add project health badges

### PR DESCRIPTION
**Pull request by peternied**
### Description
Adding badges indicative of project health 

### Issues Resolved
- Related https://github.com/opensearch-project/opensearch-build/issues/2629
- Related https://github.com/opensearch-project/security/pull/2167

### Check List
- [ ] ~New functionality includes testing.~
  - [ ] ~All tests pass~
- [ ] ~New functionality has been documented.~
  - [ ] ~New functionality has javadoc added~
- [X] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
 [![Documentation](https://img.shields.io/badge/documentation-reference-blue)](https://opensearch.org/docs/latest/opensearch/index/)
-[![codecov](https://codecov.io/gh/opensearch-project/OpenSearch/branch/main/graph/badge.svg)](https://codecov.io/gh/opensearch-project/OpenSearch)
-[![GHA gradle check](https://github.com/opensearch-project/OpenSearch/actions/workflows/gradle-check.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/gradle-check.yml)
-[![GHA validate pull request](https://github.com/opensearch-project/OpenSearch/actions/workflows/wrapper.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/wrapper.yml)
-[![GHA precommit](https://github.com/opensearch-project/OpenSearch/actions/workflows/precommit.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/precommit.yml)
-[![Jenkins gradle check job](https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fbuild.ci.opensearch.org%2Fjob%2Fgradle-check%2F&label=Jenkins%20Gradle%20Check)](https://build.ci.opensearch.org/job/gradle-check/)
```
Comment (peternied) [inline README.md:9]: Do we want to keep these badges? They didn't seem as actionable for the prime real estate on this page.

File: README.md [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
+[![](https://img.shields.io/github/issues/opensearch-project/opensearch/security%20vulnerability?labelColor=red)](https://github.com/opensearch-project/opensearch/issues?q=is%3Aissue+is%3Aopen+label%3A"security%20vulnerability")
+[![](https://img.shields.io/github/issues/opensearch-project/opensearch)](https://github.com/opensearch-project/opensearch/issues)
+[![](https://img.shields.io/github/issues-pr/opensearch-project/opensearch)](https://github.com/opensearch-project/opensearch/pulls)
+[![](https://img.shields.io/codecov/c/gh/opensearch-project/opensearch)](https://app.codecov.io/gh/opensearch-project/opensearch)
+[![](https://img.shields.io/github/issues/opensearch-project/opensearch/v2.4.0)](https://github.com/opensearch-project/opensearch/issues?q=is%3Aissue+is%3Aopen+label%3A"v2.4.0")
+[![](https://img.shields.io/github/issues/opensearch-project/opensearch/v3.0.0)](https://github.com/opensearch-project/opensearch/issues?q=is%3Aissue+is%3Aopen+label%3A"v3.0.0")
 
 - [Welcome!](#welcome)
```
Comment (peternied) [inline README.md:13]: [Click here](https://github.com/peternied/OpenSearch-1/blob/readme-badges/README.md) to see the readme with this changes.

File: README.md [hunk 3]
```diff
@@ -2,11 +2,13 @@
 
 [![Chat](https://img.shields.io/badge/chat-on%20forums-blue)](https://forum.opensearch.org/c/opensearch/)
 [![Documentation](https://img.shields.io/badge/documentation-reference-blue)](https://opensearch.org/docs/latest/opensearch/index/)
-[![codecov](https://codecov.io/gh/opensearch-project/OpenSearch/branch/main/graph/badge.svg)](https://codecov.io/gh/opensearch-project/OpenSearch)
```
Comment (anasalkouz) [inline README.md:5]: I prefer to keep codecov as third badge and using this impmentation since it has prettier look and feel.

File: README.md [hunk 4]
```diff
@@ -2,11 +2,13 @@
...
-[![GHA gradle check](https://github.com/opensearch-project/OpenSearch/actions/workflows/gradle-check.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/gradle-check.yml)
-[![GHA validate pull request](https://github.com/opensearch-project/OpenSearch/actions/workflows/wrapper.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/wrapper.yml)
-[![GHA precommit](https://github.com/opensearch-project/OpenSearch/actions/workflows/precommit.yml/badge.svg)](https://github.com/opensearch-project/OpenSearch/actions/workflows/precommit.yml)
-[![Jenkins gradle check job](https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fbuild.ci.opensearch.org%2Fjob%2Fgradle-check%2F&label=Jenkins%20Gradle%20Check)](https://build.ci.opensearch.org/job/gradle-check/)
+[![](https://img.shields.io/github/issues/opensearch-project/opensearch/untriaged?labelColor=red)](https://github.com/opensearch-project/opensearch/issues?q=is%3Aissue+is%3Aopen+label%3A"untriaged")
```
Comment (anasalkouz) [inline README.md:5]: Can you add documentation for each badge similar to previous one. Example:
[![Untriaged Issues](...

File: README.md (hunk 1 above)
Comment (anasalkouz) [reply README.md:9]: True, may be those are not actionable, but may be those added as quick links similar to the documentation badge. @dreamer-89 what is your thought?

**Review by anasalkouz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dreamer-89) [reply README.md:9]: Yes, the existing badges provides status and quick links to related workflow/job. 

@dblock : what are you thoughts on this ?

Comment (dblock) [reply README.md:9]: Sounds like 2 people find them useful @peternied, I would put those back.

**Comment by peternied** [comment]
@anasalkouz @dreamer-89 What do you think?

**Review by dblock — APPROVED** [review]

**Review by anasalkouz — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (dreamer-89) [reply README.md]: Thanks @peternied, this looks good to me.

**Pull request merged**